server/area_manager.py

```diff
@@ -460,14 +461,18 @@ def get_cms(self):
         class Testimony:
             """Represents a complete group of statements to be pressed or objected to."""
             
-            def __init__(self, title):
+            def __init__(self, title, limit):
                 self.title = title
-                self.statements = [] * 30 # if you have more than 30 statements, don't.
+                self.statements = []
+                self.limit = limit
             
             def add_statement(self, message):
                 """Add a statement and return whether successful."""
                 message = message[:14] + (1,) + message[15:]
                 self.statements.append(message)
+                if len(self.statements) > self.limit:
+                    self.statements = self.statements[:self.limit]
+                    return False
```

Comment:
```suggestion
                if len(self.statements) > self.limit:
                    return False
                self.statements.append(message)
```

Reply:
Done, but I used an >= instead of an > to make sure we don't go over the statement limit.